core/input/input_event.h

```diff
@@ -56,6 +56,9 @@ class InputEvent : public Resource {
 	int device = 0;
 
 protected:
+	bool canceled = false;
```

Comment:
Let me suggest making these a bit field, or use a hand-crafted enum + flags construct. Due to alignment, maybe we wouldn't be gaining in terms of object size, though.